### PR TITLE
Fix some clippy warnings

### PR DESCRIPTION
**Pull request by remilauzier**

### CODE REVIEW

**Comment by tomassedovic** [comment]
Hi! Thank you for the PR! I've started the CI and I'll take a look later (going away for the weekend).

I'm not sure I'll be able to merge it though. Basically, I plan to release the game on Steam and possibly macOS' App Store and I seem to remember that due to the restrictions the stores place on it, you can only do that if all the copyright holders agree.

I need to read up on the actual legal situation of doing that and seeing what needs to happen. It might need to be your permission or a copyright assignment (which I don't like doing :-( ) or maybe it's all fine.

Sorry for not investigating this earlier and having a PR guideline ready! I'd understand if you want to withdraw the contribution given these potential complications.

**Comment by remilauzier** [comment]
If you need my agreements for my code to go on these store i am OK with it.
You should talk to the developers of openttd. they are on steam even as a free software maybe they know all these legal stuff.

File: src/graphic.rs [hunk 1]
```diff
@@ -168,27 +168,27 @@ pub fn tilemap_coords_px(_tilesize: u32, graphic: Graphic) -> Option<(i32, i32)>
         FoodAcornWide => Some((2 + 3, 12 - 3)),
         FoodAcornThin => Some((3 + 3, 12 - 3)),
         FoodCarrotWide => Some((1 + 3, 12 - 3)),
-        FoodCarrotSideways => Some((0 + 3, 12 - 3)),
+        FoodCarrotSideways => Some((3, 12 - 3)),
```
Comment (tomassedovic) [inline src/graphic.rs:171]: I know it's weird but I'd prefer to keep this as is.

These numbers are offsets in the tilemap position and as such I like them to keep an explicit format that's the same for every tile.

Is there a way to allow this particular warning at a block/function level?

File: src/monster.rs [hunk 2]
```diff
@@ -124,10 +124,7 @@ impl Monster {
             Signpost => Behavior::Immobile,
         };
 
-        let invincible = match kind {
-            Npc | Signpost => true,
-            _ => false,
-        };
+        let invincible = matches!(kind, Npc | Signpost);
```
Comment (tomassedovic) [inline src/monster.rs:127]: I completely forgot `matches!` is a thing now! This is a very old codebase :-) (on Rust timeframes).

**Review by tomassedovic — COMMENTED** [review]
Okay, this is awesome. Thank you so much!

For some reason, even when I allow the CI to run on your PR, it doesn't work. But when I open a new PR myself with literally the same changes, it passes. I'll figure that out later. For now I can run it manually that way.

With the exception in the `src/graphic.rs` thing, I'm happy for this to go in if you're happy with me publishing the game on Steam/App Store etc. any time in the future.

How do you want to do this? Do you plan to keep fixing the remaining clippy warnings or would you prefer I merge this now? I'm happy either way.

**Comment by remilauzier** [comment]
I correct the problem in graphic.rs and put an option to deactivate this warning. I am ok with the code going in the appstore and steam.

**Comment by remilauzier** [comment]
You can merge it when you're ready. if i find something else i just made another pr.

**Pull request merged**